### DISCUSSION TITLE
reload the file after git checks out to another branch - Enhancement

### DISCUSSION THREAD

**Discussion by hadilq** [helix-editor/helix#6791] [Ideas]
I recently switched to helix from vim. I missed some features, which the most important one is the one in the title.

1. `hx` is not loading the files after I checked out to another branch. at least I expect to see a message to check my intention: do I want to keep working on the previous changes, or do I want to see the file `as is` in the new branch?
2. I missed `[{` and `]}` in vim, which goes to the previous `{`, or next  `}`, specially when I navigate a json file. Also supporting `[[`, `[(`, etc would be nice.
3. I used to use `jj` to go to the normal mode, so I set it up in helix too, but the problem is if I type one `j` it waits for another `j` for ever,  unless I type something else and remove it to just type `j`! I expect it to type j after a second or so without any further efforts!
4. In the normal mode, if I use `w` to "Move next word start", then type `i` to go to the insert mode, because it already had selected that word, the cursor jumps to the start of the selected word, so I have to move by a character to right or left to unselect that word, then move back and type `i`! I expect to be able to enter the edit mode where the cursor is.
5. in the "Open jumplist picker" can I have the list of files base on what position I recently worked on, and also the numbering in a unique way so I could just press a number to filter it out and jump to that position?
6. in the view mode, if I want to "Scroll the view downwards," after pushing `j` once it'll go to the normal mode, but I expect to scroll as much as possible without changing the position of the cursor.

sorry for the long list in one issue, but except the first one that I expect to be fixed, the rest are just suggestions. If any of the suggestions looks okay, I'll create a specific issue for that if you ask.

**Comment by kirawi** [comment]
1: Needs a file watcher implementation which is complicated. You can use `:reload-all` to manually do it.
4: Use `a` to append.
